### PR TITLE
password gui with pinentry-mac for handling Casks that require sudo

### PR DESCRIPTION
**Pull request by swissbuechi**
Solves Issue https://github.com/Homebrew/homebrew-autoupdate/issues/40

You can now pass the argument `--sudo` to the `start` command.
If a Cask requires sudo, pinetry-mac will open a gui to ask for the password:
image

### CODE REVIEW

**Comment by swissbuechi** [comment]
The users currently need to manually run `brew install pinentry-mac`.
Should I add a check if `pinentry-mac` is installed and display a hint to the users? (Only if the `--sudo` argument is used)
Or should we auto-install the dependency if it is missing?

File: README.md [hunk 1]
```diff
@@ -67,6 +67,10 @@ brew autoupdate version:
                                    instead of waiting for one interval (24
                                    hours by default) to pass first. Must be
                                    passed with start.
+      --sudo                       If a Cask requires sudo, autoupdate will 
+                                   open a GUI to ask for the password.
+                                   Requires https://formulae.brew.sh/formula/pinentry-mac
+                                   to be installed.
```
Comment (MikeMcQuaid) [inline README.md:73]: It would be better to use `Formula["pinentry-mac"].any_version_installed?` in the code rather than documenting the requirement.

**Comment by MikeMcQuaid** [comment]
> Should I add a check if `pinentry-mac` is installed and display a hint to the users? (Only if the `--sudo` argument is used)
> Or should we auto-install the dependency if it is missing?

Up to @DomT4 but I think at least it should check in the code.

File: README.md (hunk 1 above)
Comment (swissbuechi) [reply README.md:73]: Ok, I will add a check in the code.

Comment (swissbuechi) [reply README.md:73]: I added it in https://github.com/Homebrew/homebrew-autoupdate/pull/110/commits/83b26e9e46fe5ff1e5bc1c9d5d682945a5d0a827

**Comment by swissbuechi** [comment]
> The window spawned by `pinentry-mac` steals the focus, and also you can not minimize it. I will look into the docs, maybe there is an option to enable the `minimize` button.

Focus: `--no-global-grab` option is already set 👍🏻
Minimize: This is currently not possible with `pinentry-mac`

**Comment by DomT4** [comment]
This is good work, thank you @swissbuechi. Sort of held off moving on this broader issue because I was trying to think of something both native and lightweight, and never _really_ found anything I was happy with. Agree this is at least better than the status quo and a very visible improvement to the end user.

Will give this another glance over no later than Wednesday but from quick glance it looks like a better workaround than anything I've cooked up 😄.

Thanks for the code review @MikeMcQuaid 🙌🏻

**Comment by swissbuechi** [comment]
> This is good work, thank you @swissbuechi. Sort of held off moving on this broader issue because I was trying to think of something both native and lightweight, and never _really_ found anything I was happy with. Agree this is at least better than the status quo and a very visible improvement to the end user.
> 
> Will give this another glance over no later than Wednesday but from quick glance it looks like a better workaround than anything I've cooked up 😄.
> 
> Thanks for the code review @MikeMcQuaid 🙌🏻

Thank you 👍🏻
I'm looking forward to your review.

A litte bit off-topic, but could you please unlock issue: https://github.com/Homebrew/homebrew-autoupdate/issues/59 ?
I would like to discuss this and maybe create another PR to provide an option to use `StartCalendarInterval`.
Maybe you also need to add some labels, so the bot won't auto-close it again.

Thanks.

**Comment by MikeMcQuaid** [comment]
> A litte bit off-topic, but could you please unlock issue: #59 ?

@swissbuechi done, hope that's ok @DomT4, feel free to re-lock if not and: sorry!

**Pull request merged**